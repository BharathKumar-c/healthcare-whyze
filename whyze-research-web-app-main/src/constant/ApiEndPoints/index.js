const FEASIBILITY_API = process.env.REACT_APP_FEASIBILITY_STUDY_BASE_URL;
const AUTH_API = process.env.REACT_APP_AUTH_URL;

export const feasibilityStudyApiUrls = {
  medicalConditionSearchUrl: `${FEASIBILITY_API}/search/indication`,
  medicationConditionSearchUrl: `${FEASIBILITY_API}/search/medication`,
  caseStudyUrl: `${FEASIBILITY_API}/case`,
  locationUrl: `${FEASIBILITY_API}/case/location`,
  initialPopulation: `${FEASIBILITY_API}/case/get-count`,
  allergiesListSearchUrl: `${FEASIBILITY_API}/search/allergies`,
  reactionListSearchUrl: `${FEASIBILITY_API}/search/reaction`,
  sitesDataUrl: `${FEASIBILITY_API}/case/site`,
  sitesWithPatientsCountUrl: `${FEASIBILITY_API}/case/site-patient-count`,
};

export const authEndPoints = {
  login: `${AUTH_API}/login`,
  register: `${AUTH_API}/register`,
  refreshToken: `${AUTH_API}/refreshToken`,
  forgotPassword: `${AUTH_API}/sendResetPasswordLink`,
  resetPassword: `${AUTH_API}/resetPassword`,
};

export const dashboardApiUrls = {
  projectUrl: `${FEASIBILITY_API}/project`,
  favouriteUrl: `${FEASIBILITY_API}/project/favourite`,
};

export const profileApiUrls = {
  image: `${AUTH_API}/image`,
  user: `${AUTH_API}/user`,
  profilePicture: `${AUTH_API}/profilePicture`,
};
