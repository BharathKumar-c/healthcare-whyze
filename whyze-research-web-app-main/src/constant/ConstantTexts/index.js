import React from 'react';

// dashboard welcome Screen const text

export const welComeComponentContent =
  'See how Whyze is preforming for researchers like you.';
export const createFirstStudy = 'Create your first study';
export const createFirstStudyContent =
  'Define your protocol or create your first feasibility study';
export const patientstitleLable = 'Patients';
export const totalPatientsLable = 'Total patients';
export const countriesLable = 'Countries';
export const sitesLable = 'Sites';
export const sitesRecruitedLable = 'Sites recruited';
export const avgTimeLable = 'Avg. response time';
export const trialsLable = 'Trials';
export const activeTrials = 'Active trials';
export const totalTrials = 'Total trials';
export const screenFailRate = 'Screen fail rate';
export const threeMonths = '3 months';
export const avgTimeToRecruit = 'Avg. time to recruit';
export const feasibilityStudiesInPlanning = 'Feasibility studies in planning';
export const seeAll = 'See all';

//Profile Page const text
export const ResearcherTitle = 'Researcher title -';
export const AuthProfile = 'AUTHENTICATED PROFILE';
export const ContactDetails = 'Contact details';
export const CompanyEmail = 'Company email address';
export const PhoneNumber = 'Phone';

//ContactModal Page Const texts

export const EmailAddress = 'Email address';
export const Phone = 'Phone';
export const okButtonLabel = 'Ok';
export const saveButtonLabel = 'save';
export const cancelButtonLabel = 'cancel';
export const contactDetailsLabel = 'Contact details';

//Addcriteria texts
export const addCriteriaTitle = 'Select a criteria item';

//RenameModal texts
export const renameModalTitle = 'Profile details';
export const fullNameLabel = 'Full name';
export const titleLabel = 'Title';

// Medication constant texts
export const medicationConstants = {
  medicationLabel: 'Medication',
  medicationErrorMessage: 'Please add the Medication',
  selectlabel: 'Select Dosage',
  inputLabel: 'Enter value',
  startedPrefixText: 'Started at least',
};
// Indication constant texts
export const indicationConstants = {
  medicalConditionLabel: 'Medical condition',
  medicalConditionErrorMessage: 'Please add the Medical condition',
};

//Allergies constant texte
export const allergiesConstants = {
  allergiesLabel: 'Allergies criteria',
  allergiesPlaceholder: 'Allergies',
  allergiesErrorText: 'Please add the Allergies',
};
//Symptom constant texts

export const symptomConstats = {
  symptomLabel: 'Symptom criteria',
  symptomPlaceholder: 'Symptom',
};
//Project settingmodal Constants

export const projectSettingConstants = {
  addProjectError: 'Please add the project',
  addProjectLabel: 'Project',
  addClientNameError: 'Please add ClientName',
  rangeLabel: 'Select',
  addClientLabel: 'Client name',
  description: 'Description',
  sharingLabel: 'Sharing',
  shareBtnLabel: 'Share',
  standardsText: 'Standards',
  standardDescription: 'Standardising markers and value outputs',
  measurementText: 'Default measurement',
  clinicalTrialError: 'Please add the clinical trial',
  clinicalTrial: 'Clinical Trial',
  sponsorError: 'Please add the sponsor',
  sponsor: 'Sponsor',
  projectSettings: 'Project settings',
  descriptionError: 'Please add the description',
  standardsError: 'Please select the standards',
  noIndication: 'No Indication',
};
export const shareProjectConstants = {
  addPeoplePlaceHolder: 'Add people and groups',
  accessPeopleText: 'People with access',
};
export const selectedPeoplesConstants = {
  addPeople: 'Add people and groups',
  notifyPeopleLabel: 'Notify people',
  messageLabel: 'Message',
};
// Nicotine constant texts
export const nicotineConstants = {
  nicotineConditionErrorMessage: 'Please add the nicotine condition',
  nicotineLabel: 'Nicotine criteria',
};

// Create Account constant texts
export const createAccountContants = {
  createAccounTitle: (
    <>
      Set up your <br /> Whyze account
    </>
  ),
  setUpWhyzeAccount: 'Set up your Whyze account',
  nameLabel: 'Name',
  emailLabel: 'Company email',
  phoneLabel: 'Your phone number',
  passwordLabel: 'Create password',
  characterLengthLabel: 'Between 8 and 20 characters',
  numericLabel: 'At least one number',
  uppercaseLabel: 'At least one upper-case letter',
  lowercaseLabel: 'At least one lower-case letter',
  createAccountButtonLabel: 'Create account',
  agreeConditionText: 'By clicking the button above you agree to our',
  termsOfUseText: 'Terms of use',
  privacyNoticesText: 'Privacy policies',
  alreadyHaveAccountText: 'Already have an account?',
  loginText: 'Login',
  termsOfUseLink: 'https://www.whyzehealth.com/privacy-notice/',
  privacyNoticesLink: 'https://www.whyzehealth.com/terms',
};

//indication const

export const conditions = {
  started: 'started',
  dosage: 'dosage',
  courseOfTherapy: 'course of therapy',
  diagnosed: 'diagnosed',
  recovered: 'recovered',
  severity: 'severity',
};

//demographics const

export const demographicsConst = {
  age: 'age',
  min: 'min',
  max: 'max',
  gender: 'gender',
};

//sites const

export const siteConst = {
  siteInformationCosnt: {
    patientSegmentHeader: 'Patient segment',
    connectedWithSite: 'Connected with this site',
    additionalCatchmentAreas:
      'Additional Whyze patients in site catchment areas',
    sitePerformance: 'Site performance',
    recruitmentTimeline: 'Recruitment timeline',
    dropOutRate: 'Drop out rate',
    trackRecordOfDoingTrials: 'Track record of doing trials',
    completedInThePastyear: 'Completed in the past year',
    primaryInvestigator: 'Primary Investigator',
    administration: 'Administration',
    screenFailRate: 'Screen fail rate',
    industryAverage: 'Industry average',
    avgResponseTime: 'Avg. response time',
  },
};

//setupAccount const

export const setupAccountConst = {
  dontHaveaccount: 'Don’t have an account yet?',
  setUpAccount: 'Set up account',
};

//forgotpassword const

export const forgotpasswordConst = {
  WHYZE: 'WHYZE',
  forgotPassword: 'Forgot Password',
  resendEmail: 'Enter your email below and we will send you a reset email.',
  pleaseTry: 'Didn’t receive it ? Please try again',
  returnTo: 'return to',
  loginPage: 'login',
};

//weight const  texts

export const weightConst = {
  weightRange: 'Weight Range',
  weightMin: 'min',
  weightMax: 'max',
  weightType: 'number',
  weightmaxErrorText: 'add max weight',
  weightminErrorText: 'add min weight',
  weightFrom: 'From',
  weightTo: 'To',
};

export const SaveAndCloseModalConst = {
  SaveAndCloseModalTitle: 'Name before exiting',
  projectNameInputLabel: 'Project',
  projectNameInputPlaceholder: 'New patient segment',
  clientNameInputLabel: 'Client name',
  descriptionNameInputLabel: 'Description',
  patientSegmentInputLabel: 'Patient segment required',
};

// Proposed Site list constant texts

export const proposedSiteListConstants = {
  title: 'Proposed Site List',
  seeMore: 'See more',
  emptyContent:
    'To create your proposed list, drag & drop or add your selected sites here.',
};

// Surgery constant text
export const surgeryConstants = {
  surgeryInputLabel: 'Surgery criteria',
  surgeryInputErrorMessage: 'Please add the Surgery',
};

export const feasibilityStudy = {
  newFeasibilityStudy: 'new-feasibility-study',
};
