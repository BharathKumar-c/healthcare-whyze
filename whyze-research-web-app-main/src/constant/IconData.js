import React from 'react';

import {
  Alcohol,
  Allergies,
  Availability,
  Bmi,
  Indication,
  Medication,
  Previousstudy,
  Surgery,
  Symptom,
  Weight,
  Nicotine,
  Demographics,
  LocationIcon,
} from '../assets';

export const IconList = [
  {
    Id: 1,
    Title: 'Indication',
    Icons: <Indication />,
  },
  { Id: 2, Title: 'Medication', Icons: <Medication /> },
  {
    Id: 3,
    Title: 'Surgery',
    Icons: <Surgery />,
  },
  {
    Id: 4,

    Title: 'Symptom',
    Icons: <Symptom />,
  },
  {
    Id: 5,
    Title: 'Alcohol',
    Icons: <Alcohol />,
  },
  {
    Id: 6,
    Title: 'Allergies',
    Icons: <Allergies />,
  },
  {
    Id: 7,
    Title: 'BMI',
    Icons: <Bmi />,
  },
  {
    Id: 8,
    Title: 'Nicotine',
    Icons: <Nicotine />,
  },
  {
    Id: 9,
    Title: 'Weight',
    Icons: <Weight />,
  },

  {
    Id: 10,

    Title: 'Availability',
    Icons: <Availability />,
  },
  {
    Id: 11,

    Title: 'Previous study',
    Icons: <Previousstudy />,
  },
  {
    Id: 12,
    Title: 'Demographics',
    Icons: <Demographics />,
  },
  {
    Id: 13,
    Title: 'Location',
    Icons: <LocationIcon />,
  },
];
