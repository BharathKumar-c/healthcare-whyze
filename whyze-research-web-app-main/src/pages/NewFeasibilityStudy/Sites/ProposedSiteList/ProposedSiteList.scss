@import '../../../../common_styles/common_variable.scss';

.proposed-site-list {

    &-title {
        display: flex;
        align-items: center;
        margin-bottom: 15px;
        @include font(700, 20px, 130%, $primary-text-color);
    }

    &-container {
        background: #FDFDFD;
        border-radius: 12px;
        padding: 16px;
    }

    &-icon {
        path {
            fill: $secondary-text-color;

        }

        padding: 0 0.2rem 0 0
    }

    &-text {
        @include font(400, 16px, 140%, $secondary-text-color);

        &-active {
            @include font(400, 16px, 140%, $primary-text-color);
        }
    }

    &-seemore {
        &-button {
            display: flex !important;
            align-items: center;
            justify-content: flex-end;

            path {
                fill: rgba(45, 51, 57, 0.45);
            }

            &-active {
                display: flex !important;
                align-items: center;
                justify-content: flex-end;
                cursor: pointer;

                path {
                    fill: $primary-color;
                }
            }
        }

        &-text {
            @include font(700, 12px, 16px, rgba(45, 51, 57, 0.45) !important);

            &-active {
                @include font(700, 12px, 16px, $primary-color !important);
            }
        }
    }

    &-empty {
        &-container {
            height: 200px;
            display: flex;
            align-items: center;
            text-align: center;
            background-image: url('../../../../assets/EmptyContentBackground.png');
            background-repeat: no-repeat;
            background-position: center;
            background-size: contain;
        }

        &-text {
            @include font(400, 16px, 140%, $primary-text-color);
        }
    }

    &-sites {
        &-container {
            margin-bottom: 8px;

            &:first-child {
                margin-top: 20px;
            }
        }

        &-content {
            padding: 8px 11px;
            background: #FDFDFD;
            border: 1px solid $porcelain-color;
            border-radius: 8px;
        }
    }

    &-site-name {
        @include font(400, 16px, 140%, $primary-text-color);
        overflow: hidden;
        white-space: nowrap;
        text-overflow: ellipsis;
    }

    &-close-icon {
        display: flex !important;
        justify-content: center;
        align-items: center;

        path {
            fill: $primary-color;
            cursor: pointer;
        }
    }
}