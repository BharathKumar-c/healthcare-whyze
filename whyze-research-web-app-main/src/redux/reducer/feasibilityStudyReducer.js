import { createSlice } from '@reduxjs/toolkit';
import { v4 as uuid } from 'uuid';

export const initialState = {
  segments: [
    {
      name: 'Segment-1',
      isDone: false,
      activeTab: {
        name: 'feasibility',
        index: 0,
      },
      tabs: [
        {
          name: 'feasibility',
          title: 'Feasibility',
          isActive: true,
          position: 'left',
        },
      ],
      feasibilityStudy: [
        {
          name: 'Indication',
          case: 'Indication',
          condition: 'Inclusion',
          caseName: '',
          child: [],
        },
        {
          name: 'Demographics',
          case: 'Demographics',
          condition: 'Inclusion',
          caseName: '',
          child: [
            {
              id: uuid().slice(0, 8),
              condition: 'Gender',
              field: 'gender',
              gender: '',
            },
            {
              id: uuid().slice(0, 8),
              condition: 'Age range',
              field: 'age',
              ageRange: {
                min: 0,
                max: 0,
              },
            },
          ],
        },
      ],
      country: [],
      location: [],
      selectedCountries: [],
      totalConnectedNotConnected: {},
      sites: [],
      sitesContacted: [],
      sitesInitiated: [],
      preferredSites: [],
      patientCount: {
        feasibility: 0,
        location: 0,
      },
      siteCount: {
        feasibility: 0,
        location: 0,
      },
    },
  ],
  cardValue: {
    case: 'Indication',
    condition: '',
    index: 0,
  },
  activeSegment: {
    name: '',
    index: 0,
  },
};

const feasibilityStudySlice = createSlice({
  name: 'feasibilityStudyAction',
  initialState: initialState,
  reducers: {
    setCardDetails: (state, { payload }) => {
      state.cardValue = payload;
    },
    setSegments: (state, { payload }) => {
      state.segments = payload;
    },
    setActiveSegment: (state, { payload }) => {
      state.activeSegment = payload;
    },
    resetSegments: (state) => {
      state.segments = initialState.segments;
      state.cardValue = {
        case: 'Indication',
        condition: '',
        index: 0,
      };
      state.activeSegment = {
        name: '',
        index: 0,
      };
    },
  },
});

export const { setCardDetails, setSegments, setActiveSegment, resetSegments } =
  feasibilityStudySlice.actions;

export default feasibilityStudySlice.reducer;
