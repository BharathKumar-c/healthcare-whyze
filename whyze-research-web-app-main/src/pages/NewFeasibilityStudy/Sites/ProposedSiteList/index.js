import React from 'react';
import { Col, Row } from 'antd';
import { Text } from '../../../../components/basic';
import {
  CloseIcon,
  RightArrow,
  Sites,
  UserAdd,
  UserDelete,
  Vector,
} from '../../../../assets';
import './ProposedSiteList.scss';
import { useSelector, useDispatch } from 'react-redux';
import { setSegments } from '../../../../redux/reducer/feasibilityStudyReducer';
import { proposedSiteListConstants } from '../../../../constant/ConstantTexts';

export default function ProposedSiteList() {
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );

  const dispatch = useDispatch();
  const { preferredSites } = segments[activeSegment.index];

  let connectedSites = preferredSites.reduce((a, b) => {
    return a + b.connected;
  }, 0);
  let notConnectedSites = preferredSites.reduce((a, b) => {
    return a + b.not_connected_sites;
  }, 0);

  const handleSeeMore = () => {
    const newSegments = JSON.parse(JSON.stringify(segments));
    newSegments[activeSegment.index]?.tabs?.forEach((ele, index) => {
      if (ele.isActive) {
        ele.isActive = false;
        ele.position = 'left';
      } else if (index === 3) {
        ele.isActive = true;
      }
    });
    if (newSegments[activeSegment.index]?.tabs?.length === 4) {
      newSegments[activeSegment.index].activeTab = {
        name: newSegments[activeSegment.index].tabs[
          newSegments[activeSegment.index]?.activeTab?.index + 1
        ].name,
        index: newSegments[activeSegment.index]?.activeTab?.index + 1,
      };
    } else {
      newSegments[activeSegment.index].tabs.push({
        name: 'preferredList',
        title: 'PreferredList',
        isActive: true,
        position: 'left',
      });
      newSegments[activeSegment.index].activeTab = {
        name: 'preferredList',
        index: 3,
      };
    }
    dispatch(setSegments(newSegments));
  };

  const handleRemoveSite = (index) => {
    const newSegments = JSON.parse(JSON.stringify(segments));
    const newSites = [...preferredSites];
    newSites.splice(index, 1);
    newSegments[activeSegment.index].preferredSites = [...newSites];
    dispatch(setSegments(newSegments));
  };

  return (
    <div className="proposed-site-list">
      <Text
        className="proposed-site-list-title"
        text={proposedSiteListConstants.title}
      />
      <div className="proposed-site-list-container">
        <Row>
          <Col span={14}>
            {preferredSites?.length === 0 ? (
              <Row>
                <Col span={8}>
                  <Row>
                    <Col span={10}>
                      <Vector
                        height={23}
                        width={18}
                        className="proposed-site-list-icon"
                      />
                    </Col>
                    <Col className="proposed-site-list-text" span={14}>
                      00
                    </Col>
                  </Row>
                </Col>
                <Col span={8}>
                  <Row>
                    <Col span={10}>
                      <Sites className="proposed-site-list-icon" />
                    </Col>
                    <Col className="proposed-site-list-text" span={14}>
                      00
                    </Col>
                  </Row>
                </Col>
              </Row>
            ) : (
              <Row>
                <Col span={8}>
                  <Row>
                    <Col span={10}>
                      <Sites
                        height={17}
                        width={23}
                        className="proposed-site-list-icon"
                      />
                    </Col>
                    <Col className="proposed-site-list-text-active" span={14}>
                      {preferredSites?.length || 0}
                    </Col>
                  </Row>
                </Col>
                <Col span={8}>
                  <Row>
                    <Col span={10}>
                      <UserAdd className="proposed-site-list-icon" />
                    </Col>
                    <Col className="proposed-site-list-text-active" span={14}>
                      {connectedSites || 0}
                    </Col>
                  </Row>
                </Col>
                <Col span={8}>
                  <Row>
                    <Col span={10}>
                      <UserDelete className="proposed-site-list-icon" />
                    </Col>
                    <Col className="proposed-site-list-text-active" span={14}>
                      {notConnectedSites || 0}
                    </Col>
                  </Row>
                </Col>
              </Row>
            )}
          </Col>
          <Col
            className={
              preferredSites.length === 0
                ? 'proposed-site-list-seemore-button'
                : 'proposed-site-list-seemore-button-active'
            }
            span={10}
            align="end"
            onClick={preferredSites.length === 0 ? null : handleSeeMore}
          >
            <Text
              className={
                preferredSites.length === 0
                  ? 'proposed-site-list-seemore-text'
                  : 'proposed-site-list-seemore-text-active'
              }
              text={proposedSiteListConstants.seeMore}
            />
            <RightArrow />
          </Col>
        </Row>
        {preferredSites.length === 0 ? (
          <div className="proposed-site-list-empty-container">
            <Text
              className="proposed-site-list-empty-text"
              text={proposedSiteListConstants.emptyContent}
            />
          </div>
        ) : (
          <div>
            {preferredSites.map((site, index) => (
              <Row key={index} className="proposed-site-list-sites-container">
                <Col span={20}>
                  <Row className="proposed-site-list-sites-content">
                    <Col span={4}>
                      <Sites className="proposed-site-list-icon" />
                    </Col>
                    <Col className="proposed-site-list-site-name" span={20}>
                      {site.name}
                    </Col>
                  </Row>
                </Col>
                <Col className="proposed-site-list-close-icon" span={4}>
                  <CloseIcon onClick={() => handleRemoveSite(index)} />
                </Col>
              </Row>
            ))}
          </div>
        )}
      </div>
    </div>
  );
}
