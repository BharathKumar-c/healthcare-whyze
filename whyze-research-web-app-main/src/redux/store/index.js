import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
