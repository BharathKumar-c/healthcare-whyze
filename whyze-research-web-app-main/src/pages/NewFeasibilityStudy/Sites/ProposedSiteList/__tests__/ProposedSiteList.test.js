import React from 'react';
import { act, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import userEvent from '@testing-library/user-event';

import store from '../../../../../redux/store';
import ProposedSiteList from '..';
import { proposedSiteListConstants } from '../../../../../constant/ConstantTexts';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';

function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText };
}

const proposedSiteListdata = [
  {
    id: 6,
    university: 'The Mater Misericordiae University Hospital',
    isLinked: true,
    isChecked: false,
    linkedData: [
      {
        name: 'The Mater Private',
        noOfConnectedSites: 41,
        noOfNotConnectedSites: 2,
        star: 3.3,
        location: 'Dublin, Ireland',
      },
      {
        name: 'The Mater Private',
        noOfConnectedSites: 49,
        noOfNotConnectedSites: 2,
        star: 3.2,
        location: 'Cork, Ireland',
      },
    ],
    noOfLinks: 2,
    noOfConnectedSites: 231,
    noOfNotConnectedSites: 17,
    star: '3.0',
    location: 'Dublin, Ireland',
  },
  {
    id: 7,
    university: 'Hammersmith Hospital',
    isLinked: false,
    isChecked: false,
    noOfConnectedSites: 183,
    noOfNotConnectedSites: 12,
    star: '4.2',
    location: 'London, UK',
  },
  {
    id: 8,
    university: 'University College Hospital at Westmoreland Street',
    isLinked: false,
    isChecked: false,
    noOfConnectedSites: 87,
    noOfNotConnectedSites: 6,
    star: '4.1',
    location: 'London, UK',
  },
  {
    id: 9,
    university: 'Cork University Hospital',
    isLinked: true,
    isChecked: false,
    linkedData: [
      {
        name: 'Cork University',
        noOfConnectedSites: 35,
        noOfNotConnectedSites: 11,
        star: 3.9,
        location: 'Dublin, Ireland',
      },
    ],
    noOfLinks: 1,
    noOfConnectedSites: 107,
    noOfNotConnectedSites: 7,
    star: '3.7',
    location: 'Cork, Ireland',
  },
];

describe('Test cases for proposed site list', () => {
  test('Create the snapshots for proposed site list', () => {
    const { container } = renderWithContext(<ProposedSiteList />);

    expect(container).toMatchSnapshot();
  });

  test('Should render the title correctly', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const titleText = container.getElementsByClassName(
      'proposed-site-list-title',
    );

    expect(titleText[0].textContent).toEqual(proposedSiteListConstants.title);
  });

  test('Should render the patient icon correctly', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const vectorIcon = container.getElementsByClassName(
      'proposed-site-list-icon',
    );

    expect(vectorIcon[0].textContent).toBe('Vector.svg');
  });

  test('Should show zero for the patient count correctly', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const vectorIcon = container.getElementsByClassName(
      'proposed-site-list-text',
    );

    expect(vectorIcon[0].textContent).toBe('00');
  });

  test('Should render the site icon correctly', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const siteIcon = container.getElementsByClassName(
      'proposed-site-list-icon',
    );

    expect(siteIcon[1].textContent).toBe('Sites.svg');
  });

  test('Should show zero for  the sites count correctly', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const vectorIcon = container.getElementsByClassName(
      'proposed-site-list-text',
    );

    expect(vectorIcon[1].textContent).toBe('00');
  });

  test('Should render the see more text correctly', () => {
    renderWithContext(<ProposedSiteList />);
    const seeeMoreText = screen.getByText(proposedSiteListConstants.seeMore);
    expect(seeeMoreText).toHaveClass('proposed-site-list-seemore-text');
  });

  test('Check the see more onclick function', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const seeMoreText = container.getElementsByClassName(
      'proposed-site-list-seemore-button',
    );
    expect(seeMoreText[0]).toBeInTheDocument();
  });

  test('Should render the text for empty proposed site list', () => {
    renderWithContext(<ProposedSiteList />);
    const emptyText = screen.getByText(proposedSiteListConstants.emptyContent);

    expect(emptyText).toBeInTheDocument();
  });

  test('Update Proposed site list', () => {
    const { store } = renderWithContext(<ProposedSiteList />);

    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(store.getState().feasibilityStudyReducer),
      );

      segments[activeSegment.index].preferredSites = proposedSiteListdata;
      store.dispatch(setSegments(segments));
    });
  });

  test('Should render the Sites icon correctly', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const siteIcon = container.getElementsByClassName(
      'proposed-site-list-icon',
    );

    expect(siteIcon[0].textContent).toContain('Sites.svg');
  });

  test('Should show the No of Sites correctly', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const noOfSites = container.getElementsByClassName(
      'proposed-site-list-text-active',
    );

    expect(noOfSites[0].textContent).toEqual('4');
  });

  test('Should render the user add icon correctly', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const userAddIcon = container.getElementsByClassName(
      'proposed-site-list-icon',
    );

    expect(userAddIcon[1].textContent).toContain('UserAdd.svg');
  });

  test('Should show the No of connect patient count correctly', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const noOfConnectedSites = container.getElementsByClassName(
      'proposed-site-list-text-active',
    );

    expect(noOfConnectedSites[1].textContent).toEqual('608');
  });

  test('Should render the user delete icon correctly', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const userDeleteIcon = container.getElementsByClassName(
      'proposed-site-list-icon',
    );

    expect(userDeleteIcon[2].textContent).toContain('UserDelete.svg');
  });

  test('Should show the No of connect patient count correctly', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const noOfNotConnectedSites = container.getElementsByClassName(
      'proposed-site-list-text-active',
    );

    expect(noOfNotConnectedSites[2].textContent).toEqual('42');
  });

  test('Check sites to proposed sites page tab', () => {
    const { container } = renderWithContext(<ProposedSiteList />);

    const seeMoreBotton = container.getElementsByClassName(
      'proposed-site-list-seemore-button-active',
    );
    userEvent.click(seeMoreBotton[0]);

    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(store.getState().feasibilityStudyReducer),
      );

      expect(segments[activeSegment.index].activeTab).toEqual({
        name: 'preferredList',
        index: 3,
      });
    });

    expect(seeMoreBotton[0].textContent).toContain('RightArrow.svg');
  });

  test('Should render the see more text with active style correctly', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const seeMoreText = container.getElementsByClassName(
      'proposed-site-list-seemore-text-active',
    );

    expect(seeMoreText[0]).toBeInTheDocument();
  });

  test('Add the tabs in feasibilitystudy', () => {
    const { store } = renderWithContext(<ProposedSiteList />);
    const tabs = [
      {
        name: 'feasibility',
        title: 'Feasibility',
        isActive: false,
        position: 'left',
      },
      {
        name: 'location',
        title: 'location',
        isActive: false,
        position: 'left',
      },
      {
        name: 'sites',
        title: 'sites',
        isActive: true,
        position: 'left',
      },
      {
        name: 'preferredList',
        title: 'preferredList',
        isActive: false,
        position: 'right',
      },
    ];

    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(store.getState().feasibilityStudyReducer),
      );

      segments[activeSegment.index].tabs = tabs;
      segments[activeSegment.index].activeTab = {
        name: 'sites',
        index: 2,
      };
      store.dispatch(setSegments(segments));
    });
  });

  test('Check sites to proposed sites tab with four tabs', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const seeMoreText = container.getElementsByClassName(
      'proposed-site-list-seemore-text-active',
    );

    userEvent.click(seeMoreText[0]);

    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(store.getState().feasibilityStudyReducer),
      );

      expect(segments[activeSegment.index].activeTab).toEqual({
        name: 'preferredList',
        index: 3,
      });
    });
  });

  test('Should render the university name correctly', () => {
    const { container } = renderWithContext(<ProposedSiteList />);

    const universitiesNames = container.getElementsByClassName(
      'proposed-site-list-site-name',
    );

    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(store.getState().feasibilityStudyReducer),
      );

      segments[activeSegment.index].preferredSites.forEach((ele, index) => {
        expect(universitiesNames[index].textContent).toEqual(ele.university);
      });
    });
  });

  test('Should render close icons correctly', () => {
    const { container } = renderWithContext(<ProposedSiteList />);
    const closeIcon = container.getElementsByClassName(
      'proposed-site-list-close-icon',
    );

    expect(closeIcon[0].textContent).toEqual('CloseIcon.svg');
  });

  test('Check the Click function in close icons', () => {
    renderWithContext(<ProposedSiteList />);
    const closeIcon = screen.getAllByText('CloseIcon.svg');

    userEvent.click(closeIcon[0]);
    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(store.getState().feasibilityStudyReducer),
      );
      expect(segments[activeSegment.index].preferredSites.length).toEqual(3);
    });
  });
});
