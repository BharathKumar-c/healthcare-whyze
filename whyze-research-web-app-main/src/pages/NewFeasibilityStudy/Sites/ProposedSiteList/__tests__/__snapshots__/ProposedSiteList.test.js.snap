// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for proposed site list Create the snapshots for proposed site list 1`] = `
<div>
  <div
    class="proposed-site-list"
  >
    <span
      class="ant-typography proposed-site-list-title"
    >
      Proposed Site List
    </span>
    <div
      class="proposed-site-list-container"
    >
      <div
        class="ant-row"
      >
        <div
          class="ant-col ant-col-14"
        >
          <div
            class="ant-row"
          >
            <div
              class="ant-col ant-col-8"
            >
              <div
                class="ant-row"
              >
                <div
                  class="ant-col ant-col-10"
                >
                  <svg
                    class="proposed-site-list-icon"
                    height="23"
                    width="18"
                  >
                    Vector.svg
                  </svg>
                </div>
                <div
                  class="ant-col ant-col-14 proposed-site-list-text"
                >
                  00
                </div>
              </div>
            </div>
            <div
              class="ant-col ant-col-8"
            >
              <div
                class="ant-row"
              >
                <div
                  class="ant-col ant-col-10"
                >
                  <svg
                    class="proposed-site-list-icon"
                  >
                    Sites.svg
                  </svg>
                </div>
                <div
                  class="ant-col ant-col-14 proposed-site-list-text"
                >
                  00
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          align="end"
          class="ant-col ant-col-10 proposed-site-list-seemore-button"
        >
          <span
            class="ant-typography proposed-site-list-seemore-text"
          >
            See more
          </span>
          <svg>
            RightArrow.svg
          </svg>
        </div>
      </div>
      <div
        class="proposed-site-list-empty-container"
      >
        <span
          class="ant-typography proposed-site-list-empty-text"
        >
          To create your proposed list, drag & drop or add your selected sites here.
        </span>
      </div>
    </div>
  </div>
</div>
`;
